quantum/rgblight.c

```diff
@@ -885,6 +902,21 @@ void rgblight_task(void) {
 
 #endif /* RGBLIGHT_USE_TIMER */
 
+#ifdef RGBLIGHT_USE_PROCESS
+bool process_rgblight(uint16_t keycode, keyrecord_t *record) {
+  #ifdef RGBLIGHT_IDLE_ENABLE
+    if (record->event.pressed) {
+      if (rgblight_idle_timedout) {
+        rgblight_idle_timedout = false;
+        rgblight_enable_noeeprom();
```

Comment:
Don't change the enable/disable state, as a user could have previously disabled the rgblight, and this would turn it back on against the users wishes.

Reply:
That makes sense. I will look at the rgb matrix code.

Reply:
the rgb_matrix code uses a g_suspend_state boolean locally to determine if it's suspended or not, and renders based on that.